File: Difficulty: Medium/Kadane's Algorithm/kadanes-algorithm.cpp
class Solution {
  public:
    int maxSubarraySum(vector<int> &arr) {
        // Code here
        int n = arr.size();
        int i = 1;
        
        int sum = arr[0];
        int ans = sum;
        
        while(i<n) {
            if(sum<0) sum = 0;
            sum+=arr[i];
            ans = max(ans,sum);
            i++;
        }
        
        return ans;
    }
};

File: Difficulty: Medium/Kadane's Algorithm/kadanes-algorithm.java
class Solution {
    int maxSubarraySum(int[] arr) {
        // Code here
        int n = arr.length;
        
        int j = 0;
        int sum = 0;
        
        int ans = arr[0];
        
        while(j<n) {
            
            sum+=arr[j];
            
            
            
            ans = Math.max(ans,sum);
            
            if(sum<0) sum = 0;
            j++;
        }
        
        return ans;
    }
}
